packages/sycamore-core/src/render.rs

```diff
@@ -374,3 +370,219 @@ pub fn reconcile_fragments<G: GenericNode>(parent: &G, a: &mut [G], b: &[G]) {
         }
     }
 }
+
+fn remove_old_nodes<G: GenericNode>(
+    a_slice: &[G],
+    to_keep: &Option<AHashMap<G, usize>>,
+    parent: &G,
+) {
+    for node in a_slice {
+        if !to_keep
+            .as_ref()
+            .map(|map| map.contains_key(node))
+            .unwrap_or_default()
+        {
```

Comment:
Using a combination of `!` and `unwrap_or_default()` really makes this even more confusing (although it was not great to begin with).
```suggestion
        if to_keep.is_none() || !to_keep.as_ref().unwrap().contains_key(node) {
```